### ISSUE TITLE
Minor Issues in Installation

### ISSUE THREAD

**Issue by ian-coccimiglio** [will-yx/CRISP-CODEX-Processor#1]
I was working with CRISP-CODEX, and I wanted to highlight some fixes for problems in initializing the CRISP-CODEX GUI app.
**Notes**
1. The newest version of the CUDA toolkit will not work with this software, and prompts an error message in resolving CRISP.dll within the edge_alignment package. Make sure to use version 11.2. If a newer version is installed, you may need to uninstall that version and reinstall the earlier one.
2. I examined some dll dependency issues by using a dll dependency walker (e.g., Dependencies from Github). Very helpful for diagnosing dependency-tree issues with the CRISP.dll file.
3. It is important to have control over the system environment path. On one of the computers, Windows 10 pathing was an issue, but a hack-y solution was to copy and paste the relevant .dll files from CUDA directly into the working directory (i.e., the cufft, cublas, cublasLT, and cusolver dll files from the CUDA directory).

**Ideas**
1.  I found that occasionally the program will request a specific conda environment, while also asking for pip installations of packages. This process tends to lead to ambiguities and dependency clashes when running the program. I think the best practice should be documented.

**Comment by will-yx** [comment]
Hi Ian, Thanks for the note. 

I've updated the readme file to encourage the use of a conda virtual environment [conda create -n CRISP python=3.9] during the install to prevent version clashes as well as some updated instructions to match. The use of a virtual environment will help with dependency issues. You may need to edit the [conda_activate_CRISP.bat] file if miniconda is installed in a different location or if a different environment name was used. Alternatively, you can activate your environment first [conda activate CRISP] and run [python CRISP_process_GUI.py].

For CUDA versions, you should be able to have the most recent CUDA version as well as 11.2 and even earlier versions. CUDA and vips need to be in your systems environment paths. If CUDA is properly installed on your computer (in the system path) you should not need to copy DLLs to the working path.

I'm not sure what your issues with windows pathing is, you may need to resolve that with your IT or system admin. In case you do not have install permissions on the computer, then your proposed solution will work but is not an ideal solution since other packages will not see those DLLs.

Please follow the updated instructions described in the readme file precisely and let me know if you are still having issues.

**Comment by ian-coccimiglio** [comment]
Hi Will,

Thanks for responding! I appreciate the work on the README. I'm mostly leaving these notes as some potential help for others. And yes, the system pathing was strange and not your responsibility as the package developer. But, I'll also reply to one important point here.

I did cross-test with the newest version of CUDA 12.0 - running on a Windows 10 machine with NVidia CUDA-compatible graphics card. It did not work. My understanding of the reason it doesn't work is that CUDA updated the versions of cufft64_10.dll and cublas64_11.dll, to cufft64_11.dll and cublas64_12.dll. I believe that the CODEX-CRISP software depends on the original specific dll versions. When I downgraded CUDA to 11.2, the program works.

See the attached documentation for CUDA 12.0 files as well as attached photo to see the discrepancy between CUDA 12.0 and what CRISP.dll expects as dependencies.
https://www.advanceduninstaller.com/NVIDIA-CUDA-Documentation-12_0-554dce038827ec25d0bf35b37ebc8b23-application.htm

![image](https://user-images.githubusercontent.com/62724583/219278211-5f663c28-c86e-4514-96a2-f8bfe1aa1c81.png)
